src/main/java/org/klodnicki/service/BookService.java

```diff
@@ -114,4 +113,28 @@ public List<String> sortOptionNames() {
         return namesList;
     }
 
+    public List<String> prepareListOfBooksTitleSortByParameter(String parameter, String title) throws
+            NotFoundInDatabaseException, SortParameterNotFoundException {
+        List<String> results = new ArrayList<>();
+        List<BookInfo> listOfBooksTitleSortByParameter = bookRepository.findBooksByTitleSortByParameter
+                (prepareParameterToDatabase(parameter), title);
+
+        if (listOfBooksTitleSortByParameter.isEmpty()) {
+            throw new NotFoundInDatabaseException(BookInfo.class);
+        }
```

Comment:
Please create a separate method for finding these books in database. Let prepare method just "prepare" books and extract logic for finding book to method that will be solely responsible for that.